### ISSUE TITLE
Discussion: Disable JS build pipeline when preview is not required

### ISSUE THREAD

**Issue by Bouncey** [freeCodeCamp/freeCodeCamp#16714]
Whilst QA'ing #16617, I noticed considerable lag whilst using the code editor. I believe this lag is introduced through the combination of the 750ms debounce and the attempt to transform the campers code on the fly.

Should we disable the build pipeline when the preview pane is not required?

@BerkeleyTrue 
cc @freeCodeCamp/moderators

**Comment by raisedadead** [comment]
Yes, that should be a good idea, unless there is a requirement on the build to be used by tests. Right?

**Comment by Bouncey** [comment]
The build is also triggered when the `Run Tests` button is pressed, so that shouldn't be an issue

**Comment by TJBarrettJR** [comment]
I am trying to learn about how FCC works so I am sorry if this makes no sense. And if it does not, please help educate me because I am looking to get more involved.

Would it be possible to put in a toggle of transforming the code on the fly or not, even if the preview pane is used? Then a button to build the preview if the toggle is off? That way if any users experience any lag when the preview pane is still used they can mitigate that. 
Then in challenges that do not need the preview pane it would automatically be off and hidden?
Maybe that is way more complex then I am making it sound, but it sounds like you are talking about building the basic functionality of that anyways. 

Again, please let me know if that makes no sense, thank you!

**Comment by raisedadead** [comment]
Hi @TJBarrettJR 

> I am trying to learn about how FCC works so I am sorry if this makes no sense. And if it does not, please help educate me because I am looking to get more involved.

That's awesome, and we would love to have you help us.

> Would it be possible to put in a toggle of transforming the code on the fly or not, even if the preview pane is used? Then a button to build the preview if the toggle is off? That way if any users experience any lag when the preview pane is still used they can mitigate that.
Then in challenges that do not need the preview pane it would automatically be off and hidden?
Maybe that is way more complex then I am making it sound, but it sounds like you are talking about building the basic functionality of that anyways.

Yes, we are talking about having this, internally by default. That is without any user need to toggle the setting.

That said, I am think what you are interested in is with a way to disable the preview, for example until you have completed the code, and would like to see the preview with a button like "Display Preview"?

If yes, that's a different issue than this thread, and we welcome you to log a feature request on this tracker.

P.S: We hangout in [Contributors Chat room](https://gitter.im/FreeCodeCamp/Contributors), where we can answer all queries.

Happy contributing!
